// chakra-color-picker/dist/chakra-color-picker.esm.js
import { useBoolean, Popover, PopoverTrigger, Button, PopoverContent, PopoverArrow, SimpleGrid } from '@chakra-ui/react';
import React, { useState } from 'react';

var defaultColors = ["gray.500", "red.500", "orange.500", "yellow.500", "green.500", "teal.500", "blue.500", "cyan.500", "purple.500", "pink.500"];
var ColorPicker = function ColorPicker(_ref) {
  var onChange = _ref.onChange,
      colors = _ref.colors,
      defaultColor = _ref.defaultColor,
      bg = _ref.bg,
      placement = _ref.placement,
      isDisabled = _ref.isDisabled;

  var _useBoolean = useBoolean(),
      isOpen = _useBoolean[0],
      setIsOpen = _useBoolean[1];

  var colorPalette = colors || defaultColors;

  var _useState = useState(defaultColor || colorPalette[0]),
      selectedColor = _useState[0],
      setSelectedColor = _useState[1];

  return React.createElement(Popover, {
    isOpen: isOpen && !isDisabled,
    onClose: setIsOpen.toggle,
    placement: placement
  }, React.createElement(PopoverTrigger, null, React.createElement(Button, {
    bg: selectedColor,
    onClick: setIsOpen.toggle,
    _hover: {
      bg: selectedColor,
      transform: "scale(1.05)"
    },
    _active: {
      bg: selectedColor
    },
    "aria-label": "color picker",
    isDisabled: isDisabled
  })), React.createElement(PopoverContent, {
    w: "auto",
    bg: bg,
    boxShadow: "md"
  }, React.createElement(PopoverArrow, null), React.createElement(SimpleGrid, {
    columns: 5,
    p: 1,
    spacing: 1
  }, colorPalette.map(function (color, index) {
    return React.createElement(Button, {
      key: "color-picker-" + color + "-" + index,
      h: "40px",
      w: "40px",
      p: 0,
      minW: "40px",
      bg: color,
      _hover: {
        bg: color,
        transform: "scale(1.05)"
      },
      _active: {
        bg: color
      },
      onClick: function onClick() {
        setIsOpen.toggle();
        setSelectedColor(color);
        onChange(color);
      }
    });
  }))));
};

export { ColorPicker };
